src/Client/Web/Bit.Client.Web.BlazorUI/Components/Pickers/BitDatePicker.razor.cs

```diff
@@ -144,6 +159,10 @@ public async Task HandleDateChoose(int dayOfWeek, int day, int month)
                 IsOpen = false;
                 BitDate date = new(currentYear,month,day,dayOfWeek);
                 selectedDate = OnSelectDate is not null ? OnSelectDate.Invoke(date): GetSelectedDateString(date);
+                if (ValueChanged.HasDelegate is true)
```

Comment:
please also check if the value has been set by checking the `ValueHasBeenSet` field. just like other 2-way bound parameters in other components.